CONTRIBUTING.md

```diff
@@ -142,6 +144,74 @@ Once your reviewer is happy, they will say `/lgtm`, which will apply the
 Your PR will be automatically merged once it has the `lgtm` and `approved`
 labels, does not have any `do-not-merge/*` labels, and all status checks (eg., rebase, tests, DCO) are positive.
 
+### Resolving Conflicts by Rebasing
+
+When submitting a pull request, it's important to make sure that it can be cleanly merged into the upstream repository. In some cases, conflicts may arise when attempting to merge a pull request, such as when changes have been made to the same file or lines of code.
+
+To resolve conflicts, we always use the `git rebase` command rather than git merge. This helps to ensure a clean and linear history of commits.
+
+Here's the process for rebasing a pull request to resolve conflicts:
+
+1. Ensure that your local repository is up to date with the upstream repository:
+
+```
+git remote add upstream <upstream_repository_url>
+git fetch upstream
+git checkout main
+git merge upstream/main
```

Comment:
Thanks @therealbobo! Just one question: why not rebasing on the upstream instead of merging?

Reply:
No problem :-)